src/adapter/etl-adapter-csv/src/Flow/ETL/Adapter/CSV/CSVExtractor.php

```diff
@@ -215,4 +211,34 @@ private function mapHeaders(array $headers) : array
             \array_keys($headers)
         );
     }
+
+    /**
+     * @param array<int, null|string> $rowData
+     *
+     * @return array<int, null|string>
+     */
+    private function normalizeRowData(array $rowData, int $headersCount) : array
+    {
+        $rowDataCount = \count($rowData);
```

Comment:
We already have this count at line 62, and it seems `$rowData` is not manipulated after that line. Can't we use it here?

Reply:
I'm actually going to extract this logic to standalone class that can be properly unit tested